packages/components/src/Chip/Chip.tsx

```diff
@@ -31,29 +33,73 @@ export const Chip = ({
   const prefix = useChildComponent(children, d => d.type === Chip.Prefix);
   const suffix = useChildComponent(children, d => d.type === Chip.Suffix);
 
+  const [labelRef, labelFullyVisible] = useInView<HTMLSpanElement>();
+  const [headingRef, headingFullyVisible] = useInView<HTMLSpanElement>();
+  const tooltipMessage = getTooltipMessage(
+    labelFullyVisible,
+    headingFullyVisible,
+    label,
+    heading,
+  );
+
   return (
-    <button
-      className={classes}
-      onClick={(ev: React.MouseEvent<HTMLButtonElement>) =>
-        onClick && onClick(value, ev)
-      }
-      tabIndex={tabIndex}
-      onKeyDown={onKeyDown}
-      aria-label={ariaLabel || label}
-      disabled={disabled}
-      role={role}
-      type="button"
-    >
-      {prefix}
-      <Typography size="base" fontWeight="medium">
-        {heading}
-      </Typography>
-      {heading && <span className={styles.chipBar} />}
-      <Typography size="base">{label}</Typography>
-      {suffix}
-    </button>
+    <Tooltip message={tooltipMessage}>
+      <button
+        className={classes}
+        onClick={(ev: React.MouseEvent<HTMLButtonElement>) =>
+          onClick && onClick(value, ev)
+        }
+        tabIndex={tabIndex}
+        onKeyDown={onKeyDown}
+        aria-label={ariaLabel || label}
+        disabled={disabled}
+        role={role}
+        type="button"
+      >
+        {prefix}
+        <div className={styles.chipContent}>
+          {heading && (
+            <>
+              <Typography size="base" fontWeight="medium">
+                {heading}
+                <span ref={headingRef} />
+              </Typography>
+              <span className={styles.chipBar} />
+            </>
+          )}
+          <Typography size="base">
+            {label}
+            <span ref={labelRef} />
+          </Typography>
+          {!labelFullyVisible && (
+            <div
+              className={styles.truncateGradient}
+              data-testid="ATL-Chip-Truncation-Gradient"
```

Comment:
Any chance we can bump this to a prop with a default of your value? I doubt anyone needs it today, but I could see someone down the road wanting to override this.

Reply:
I totally get where you're coming from on that. in fact I believe I either suggested or did the same thing at one point and someone gently suggested I not do that as per a guide we have. I'l send it to ya in a moment here